### PR TITLE
Implement query params translations

### PR DESCRIPTION
**Pull request by zverseckas**
Implementing query parameters translation. It can be enabled in a gem's config. The PR turned out to be too big though.

@laurynas

### CODE REVIEW

File: lib/mondial_relay/shipments/create/translate_params.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module MondialRelay
+  module Shipments
```
Comment (zverseckas) [inline lib/mondial_relay/shipments/create/translate_params.rb]: Similar code found in translate_params.rb:4, translate_params.rb:4 (mass = 202)

File: spec/mondial_relay/labels/create_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+RSpec.describe MondialRelay::Labels::Create, '.for' do
+  subject { described_class.for(params) }
```
Comment (zverseckas) [inline spec/mondial_relay/labels/create_spec.rb]: Identical code found in create_spec.rb:4, create_spec.rb:4 (mass = 244)

File: spec/mondial_relay/shipments/create_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+RSpec.describe MondialRelay::Shipments::Create, '.for' do
+  subject { described_class.for(params) }
```
Comment (zverseckas) [inline spec/mondial_relay/shipments/create_spec.rb]: Identical code found in create_spec.rb:4, create_spec.rb:4 (mass = 244)

File: spec/mondial_relay/labels/fetch_spec.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+RSpec.describe MondialRelay::Labels::Fetch, '.for' do
+  subject { described_class.for(params) }
```
Comment (zverseckas) [inline spec/mondial_relay/labels/fetch_spec.rb]: Similar code found in fetch_spec.rb:4, trace_spec.rb:4 (mass = 78)

File: spec/mondial_relay/shipments/trace_spec.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+RSpec.describe MondialRelay::Shipments::Trace, '.for' do
+  subject { described_class.for(params) }
```
Comment (zverseckas) [inline spec/mondial_relay/shipments/trace_spec.rb]: Similar code found in fetch_spec.rb:4, trace_spec.rb:4 (mass = 78)

File: spec/mondial_relay/labels/create_spec.rb [hunk 6]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+RSpec.describe MondialRelay::Labels::Create, '.for' do
```
Comment (zverseckas) [inline spec/mondial_relay/labels/create_spec.rb]: Metrics/BlockLength: Block has too many lines. [26/25]

File: spec/mondial_relay/labels/create_spec.rb [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+      receiver_city: 'Bordeaux',
+      receiver_postal_code: '33000',
+      receiver_country: 'FR',
+      parcel_weight: 500,
+      parcel_length: 20,
+      parcel_count: 1,
+    }
+  end
+
+  let (:query_result) { double(:query_result) }
```
Comment (zverseckas) [inline spec/mondial_relay/labels/create_spec.rb]: Lint/ParenthesesAsGroupedExpression: `(...)` interpreted as grouped expression.

File: spec/mondial_relay/labels/fetch_spec.rb [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+  let(:params) do
+    {
+      shipmet_numbers: '31236189',
+      language: 'FR',
+    }
+  end
+
+  let (:query_result) { double(:query_result) }
```
Comment (zverseckas) [inline spec/mondial_relay/labels/fetch_spec.rb]: Lint/ParenthesesAsGroupedExpression: `(...)` interpreted as grouped expression.

File: spec/mondial_relay/shipments/create_spec.rb [hunk 9]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+RSpec.describe MondialRelay::Shipments::Create, '.for' do
```
Comment (zverseckas) [inline spec/mondial_relay/shipments/create_spec.rb]: Metrics/BlockLength: Block has too many lines. [26/25]

File: spec/mondial_relay/shipments/create_spec.rb [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+      receiver_city: 'Bordeaux',
+      receiver_postal_code: '33000',
+      receiver_country: 'FR',
+      parcel_weight: 500,
+      parcel_length: 20,
+      parcel_count: 1,
+    }
+  end
+
+  let (:query_result) { double(:query_result) }
```
Comment (zverseckas) [inline spec/mondial_relay/shipments/create_spec.rb]: Lint/ParenthesesAsGroupedExpression: `(...)` interpreted as grouped expression.

File: spec/mondial_relay/shipments/trace_spec.rb [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
+  let(:params) do
+    {
+      shipment_number: '31236189',
+      language: 'FR',
+    }
+  end
+
+  let (:query_result) { double(:query_result) }
```
Comment (zverseckas) [inline spec/mondial_relay/shipments/trace_spec.rb]: Lint/ParenthesesAsGroupedExpression: `(...)` interpreted as grouped expression.

**Comment by laurynas** [comment]
Maybe instead of multiple translation interactors we could have one `Translatable` or `ParamsTranslator` module which would be included into actions and define `translated_params`. `TRANSLATIONS` constant could be defined in the interactor itself. Or could be extracted into modules too.

Just a suggestion if you would like to try modules instead of classes.

**Comment by zverseckas** [comment]
@laurynas, I like the idea of `Translatable`. It could define `translated_params` and later `translated_response` (I am planning on implementing response translation too).
```ruby
module Labels
  class Fetch < Operation
    OPERATION = :wsi3_get_etiquettes
    
    PARAM_TRANSLATIONS = {
      # ...,
    }
    
    RESPONSE_TRANSLATIONS = {
      # ...,
    }

    # run can be implemented in Operation class (in most of the operations)
  end
end
```
In this case `Fetch` looks more of a configuration than a class. You were saying about extracting these constants in separate modules. How do you imagine that? 🤔

**Comment by laurynas** [comment]
Yes I meant something like that :) Translatable could be included in base class. And yes, probably there is no need for translations per operation. 

On the other hand, if you would like to try, it  could be something like:

```ruby
module Labels::Fetch::Translations
  PARAMS = { ... }
  RESPONSE = { ... }
end
```

**Comment by zverseckas** [comment]
Yes, I like it. I will do it! :)

File: lib/mondial_relay/labels/create/translations.rb [hunk 12]
```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module MondialRelay
+  module Labels
```
Comment (zverseckas) [inline lib/mondial_relay/labels/create/translations.rb]: Similar code found in translations.rb:4, translations.rb:4 (mass = 192)

File: lib/mondial_relay/shipments/create/translations.rb [hunk 13]
```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module MondialRelay
+  module Shipments
```
Comment (zverseckas) [inline lib/mondial_relay/shipments/create/translations.rb]: Similar code found in translations.rb:4, translations.rb:4 (mass = 192)

**Pull request merged**